ActivatePubKey/__tests__/handler.test.ts

```diff
@@ -0,0 +1,525 @@
+import { contextMock } from "../../__mocks__/context.mock";
+
+const aFiscalCode = "SPNDNL80A13Y555X" as FiscalCode;
+
+const aValidRetrievedPopDocument: RetrievedLolliPopPubKeys = {
+  pubKey: toEncodedJwk(aValidJwk),
+  ttl: TTL_VALUE_AFTER_UPDATE,
+  assertionType: AssertionTypeEnum.SAML,
+  assertionRef: aValidSha256AssertionRef,
+  assertionFileName: `${aFiscalCode}-${aValidSha256AssertionRef}` as AssertionFileName,
+  status: PubKeyStatusEnum.VALID,
+  fiscalCode: aFiscalCode,
+  expiredAt: new Date(),
+  id: "1" as NonEmptyString,
+  version: 0 as NonNegativeInteger,
+  _etag: "",
+  _rid: "",
+  _self: "",
+  _ts: 0
+};
+
+const aPendingRetrievedPopDocument: RetrievedLolliPopPubKeys = {
+  ...aValidRetrievedPopDocument,
+  status: PubKeyStatusEnum.PENDING
+};
+
+const aValidRetrievedPopDocumentWithMasterAlgo = {
+  ...aValidRetrievedPopDocument,
+  assertionRef: aValidSha512AssertionRef,
+  assertionFileName: `${aFiscalCode}-${aValidSha512AssertionRef}`
+} as RetrievedValidPopDocument;
+
+const aPendingRetrievedPopDocumentWithMasterAlgo = {
+  ...aPendingRetrievedPopDocument,
+  // to let this document match the master we must change the assertionRef
+  // to a sha512 one
+  assertionRef: aValidSha512AssertionRef,
+  assertionFileName: `${aFiscalCode}-${aValidSha512AssertionRef}`
+};
+
+const upsertBlobFromTextMock = jest.spyOn(fn_commons, "upsertBlobFromText");
+
+const findLastVersionByModelIdMock = jest
+  .fn()
+  .mockImplementation(() => TE.of(O.some({})));
+
+const upsertMock = jest.fn().mockImplementation(() => TE.of({}));
+
+const lollipopPubKeysModelMock = ({
+  findLastVersionByModelId: findLastVersionByModelIdMock,
+  upsert: upsertMock
+} as unknown) as LolliPOPKeysModel;
+
+const LOLLIPOP_ASSERTION_STORAGE_CONTAINER_NAME = "assertions" as NonEmptyString;
+
+const popDocumentReaderMock = jest.fn(
+  (assertionRef: AssertionRef) =>
+    TE.of({
+      ...aRetrievedPendingLollipopPubKeySha256,
+      assertionRef: assertionRef,
+      id: `${assertionRef}-000000`,
+      version: 0
+    }) as ReturnType<PopDocumentReader>
+);
+
+const popDocumentWriterMock = jest.fn(
+  (item: NewLolliPopPubKeys) =>
+    TE.of({
+      ...aRetrievedPendingLollipopPubKeySha256,
+      ...item,
+      id: `${item.assertionRef}-000001`,
+      version: 1
+    }) as ReturnType<PopDocumentWriter>
+);
+const assertionWriterMock = jest.fn(
+  () => TE.of(true) as ReturnType<AssertionWriter>
+);
+
+var expiresAtDate = new Date(); // Now
+expiresAtDate.setDate(expiresAtDate.getDate() + 30); // Set now + 30 days as the new date
+
+const aValidPayload: ActivatePubKeyPayload = {
+  fiscal_code: aFiscalCode,
+  assertion: "an assertion" as NonEmptyString,
+  assertion_type: AssertionTypeEnum.SAML,
+  expires_at: expiresAtDate
+};
+
+describe("activatePubKey handler", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should success given valid informations when used algo != master algo", async () => {
```

Comment:
ditto

Reply:
done in https://github.com/pagopa/io-functions-lollipop/pull/13/commits/b5d5d231fff8ca9e7201d2acf445b0ddaca4a880